src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs

```diff
@@ -303,120 +372,411 @@ public void EnumerableStopsAtEndWhenEndIsFirstByteOfEmpty()
+            ReadOnlySequence<byte>.Enumerator enumerator = buffer.GetEnumerator();
+            {
+                Assert.Equal(default, enumerator.Current); 
+                Assert.True(enumerator.MoveNext());
+                ReadOnlyMemory<byte> memory = enumerator.Current;
+                Assert.True(memory.IsEmpty);
+
+                Assert.False(enumerator.MoveNext());
+            }
+            enumerator = new ReadOnlySequence<byte>.Enumerator(buffer);
+            {
+                Assert.Equal(default, enumerator.Current); 
+                Assert.True(enumerator.MoveNext());
+                ReadOnlyMemory<byte> memory = enumerator.Current;
+                Assert.True(memory.IsEmpty);
+
+                Assert.False(enumerator.MoveNext());
+            }
         }
 
         [Fact]
-        public void Create_WorksWithArray()
+        public void Empty_GetPosition()
         {
-            var buffer = new ReadOnlySequence<byte>(new byte[] { 1, 2, 3, 4, 5 });
-            Assert.Equal(buffer.ToArray(), new byte[] { 1, 2, 3, 4, 5 });
+            ReadOnlySequence<byte> buffer = ReadOnlySequence<byte>.Empty;
+            Assert.Equal(buffer.Start, buffer.GetPosition(0));
+            Assert.Equal(buffer.Start, buffer.GetPosition(0, buffer.Start));
+            Assert.Equal(buffer.Start, buffer.GetPosition(0, buffer.End));
+            //Assert.Equal(buffer.End, buffer.GetPosition(0));
+            //Assert.Equal(buffer.End, buffer.GetPosition(0, buffer.Start));
+            //Assert.Equal(buffer.End, buffer.GetPosition(0, buffer.End));
+        }
+
+        [Fact]
+        public void Empty_GetPositionPositive()
+        {
+            ReadOnlySequence<byte> buffer = ReadOnlySequence<byte>.Empty;
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(1, buffer.Start));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(1, buffer.End));
         }
 
         [Fact]
-        public void Empty_ReturnsLengthZeroBuffer()
+        public void Empty_GetPositionNegative()
         {
-            var buffer = ReadOnlySequence<byte>.Empty;
+            ReadOnlySequence<byte> buffer = ReadOnlySequence<byte>.Empty;
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(-1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(-1, buffer.Start));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => buffer.GetPosition(-1, buffer.End));
+        }
+
+        [Fact]
+        public void Empty_TryGet()
+        {
+            ReadOnlySequence<byte> buffer = ReadOnlySequence<byte>.Empty;
+            ReadOnlyMemory<byte> memory;
+
+            SequencePosition c1 = buffer.Start;
+            Assert.True(buffer.TryGet(ref c1, out memory, false));
```

Comment:
What does this method try to verify?

Reply:
It verify that TryGet implement correct logic for Empty ROS.